game-backend/src/MainController.ts

```diff
@@ -48,26 +51,54 @@ export default class MainController {
     matchUpdate: MatchUpdate
   ) => void {
     return (userId: UserId, matchUpdate: MatchUpdate): void => {
-      this.store.userContexts.userContextMap[userId].forEach(ws => {
+      this.store.userConnnections.userConnectionMap[userId].forEach(ws => {
         ws.send(JSON.stringify(matchUpdate));
       });
     };
   }
 
   private getTerminateExecutor(): () => void {
     return (): void => {
+      // ping all websocket connections
       setTimeout(() => {
+        this.removeInactiveUsers();
+
         this.gameController = new GameController(
           this.getMatchUpdateExecutor(),
           this.getTerminateExecutor()
         );
-        this.store.userContexts.userContextIds.forEach(id => {
+        this.store.userConnnections.userConnectionIds.forEach(id => {
           this.gameController.appendUser(id);
         });
       }, GAME_RESTART_TIME);
     };
   }
 
+  /**
+   * Remove users from store if there is not at least one connection alive
+   * @private
+   * @memberof MainController
+   */
+  private removeInactiveUsers(): void {
+    this.store.userConnnections.userConnectionIds.forEach(userId => {
+      const foundAliveConnections = this.store.userConnnections.userConnectionMap[
+        userId
+      ].filter((ws: WebSocketWithStatus) => {
+        return ws.isAlive;
+      });
+      if (foundAliveConnections.length <= 0) {
```

Comment:
Maybe consider using `.some()` instead of filter as it's a perfect match for these use cases :v: